lib/src/main/java/uk/gov/di/ipv/core/library/service/CiStorageService.java

```diff
@@ -100,25 +132,186 @@ public void submitMitigatingVcList(
+                        ? contraIndicatorEvidenceDto.getContraIndicator()
+                        : Collections.emptyList();
+        return ContraIndications.builder()
+                .contraIndicators(
+                        contraIndicators.stream()
+                                .map(this::mapToContraIndicator)
+                                .collect(
+                                        Collectors.toMap(
+                                                ContraIndicator::getCode, Function.identity())))
+                .build();
+    }
+
+    private ContraIndicator mapToContraIndicator(ContraIndicatorDto contraIndicatorDto) {
+        return ContraIndicator.builder()
+                .code(contraIndicatorDto.getCode())
+                .transactionIds(contraIndicatorDto.getTxn())
+                .documentId(contraIndicatorDto.getDocument())
+                .issuanceDate(Instant.parse(contraIndicatorDto.getIssuanceDate()))
+                .mitigations(mapToMitigations(contraIndicatorDto.getMitigation()))
+                .incompleteMitigations(
+                        mapToMitigations(contraIndicatorDto.getIncompleteMitigation()))
+                .build();
+    }
+
+    private List<Mitigation> mapToMitigations(List<MitigationDto> mitigationDtos) {
+        return mitigationDtos != null
+                ? mitigationDtos.stream()
+                        .map(
+                                mitigationDto ->
+                                        Mitigation.builder()
+                                                .code(mitigationDto.getCode())
+                                                .mitigatingCredentials(
+                                                        mitigationDto
+                                                                .getMitigatingCredential()
+                                                                .stream()
+                                                                .map(
+                                                                        this
+                                                                                ::mapToMitigatingCredential)
+                                                                .collect(Collectors.toList()))
+                                                .build())
+                        .collect(Collectors.toList())
+                : Collections.emptyList();
+    }
+
+    private MitigatingCredential mapToMitigatingCredential(
+            MitigationCredentialDto mitigationCredentialDto) {
+        return MitigatingCredential.builder()
+                .issuer(mitigationCredentialDto.getIssuer())
+                .id(mitigationCredentialDto.getId())
+                .validFrom(Instant.parse(mitigationCredentialDto.getValidFrom()))
+                .transactionId(mitigationCredentialDto.getTxn())
+                .build();
     }
 
     private boolean lambdaExecutionFailed(InvokeResult result) {
```

Comment:
I think, should update to the LogLambdaExecutionError(). So can understand which lambda didn't invoke for all lambdas. I could see "CI storage lambda execution failed" for all lamda.  Also, not sure but can be null InvokeResult?

Reply:
I think, can be some code refactor in this LogLambdaExecutionError(). Map can use instead of Hashmap. message.values.removeIf(Objects::isNull)

Reply:
Updated error msg with failed lambda arn to help identify.